src/pretix/static/pretixbase/js/asynctask.js

```diff
@@ -5,91 +5,105 @@ var async_task_check_url = null;
+        'processed. If this takes longer than two minutes, please contact us or go ' +
+        'back in your browser and try again.'
+    )
+};
+
+function async_task_schedule_check(context, timeout) {
     "use strict";
-    $.ajax(
-        {
-            'type': 'GET',
-            'url': async_task_check_url,
-            'success': async_task_check_callback,
-            'error': async_task_check_error,
-            'context': this,
-            'dataType': 'json'
+    async_task_timeout = window.setTimeout(function() {
+        $.ajax(
+            {
+                'type': 'GET',
+                'url': async_task_check_url,
+                'success': async_task_check_callback,
+                'error': async_task_check_error,
+                'context': context,
+                'dataType': 'json'
+            }
+        );
+    }, timeout);
+}
+
+function async_task_on_success(data) {
+    "use strict";
+    if ((async_task_is_download && data.success) || async_task_dont_redirect) {
+        waitingDialog.hide();
+        if (location.href.indexOf("async_id") !== -1) {
+            history.replaceState({}, "pretix", async_task_old_url);
         }
-    );
+    }
+    if (!async_task_dont_redirect)
+        location.href = data.redirect;
+    $(this).trigger('pretix:async-task-success', data);
 }
 
 function async_task_check_callback(data, textStatus, jqXHR) {
     "use strict";
     if (data.ready && data.redirect) {
-        if (async_task_is_download && data.success) {
-            waitingDialog.hide();
-            if (location.href.indexOf("async_id") !== -1) {
-                history.replaceState({}, "pretix", async_task_old_url);
-            }
-        }
-        location.href = data.redirect;
+        async_task_on_success.call(this, data);
         return;
-    } else if (typeof data.percentage === "number") {
-        $("#loadingmodal .progress").show();
-        $("#loadingmodal .progress .progress-bar").css("width", data.percentage + "%");
-        if (typeof data.steps === "object" && Array.isArray(data.steps)) {
-            var $steps = $("#loadingmodal .steps");
-            $steps.html("").show()
-            for (var step of data.steps) {
-                $steps.append(
-                    $("<span>").addClass("fa fa-fw")
-                        .toggleClass("fa-check text-success", step.done)
-                        .toggleClass("fa-cog fa-spin text-muted", !step.done)
-                ).append(
-                    $("<span>").text(step.label)
-                ).append(
-                    $("<br>")
-                )
-            }
-        }
     }
-    async_task_timeout = window.setTimeout(async_task_check, 250);
 
+    if (typeof data.percentage === "number") {
+        waitingDialog.setProgress(data.percentage);
+    }
+    if (typeof data.steps === "object" && Array.isArray(data.steps)) {
+        waitingDialog.setSteps(data.steps);
+    }
+    async_task_schedule_check(this, 250);
```

Comment:
I know originally it was checked every 250ms. But isn’t that a bit often? 4 times per second? Wouldn’t 1000ms be enough, maybe 500ms?